### PR TITLE
Enable TorchBind tests on ROCm

### PR DESCRIPTION
**Pull request by t-vi**
The torchbind tests didn't work be cause somehow we missed the rename of caffe2_gpu to torch_... (hip for us) in #20774 (merged 2019-06-13, oops) and still tried to link against it.

### CODE REVIEW

**Comment by t-vi** [comment]
@jeffdaily Lot's of debugging, tiny fix. :slightly_smiling_face:

**Comment by t-vi** [comment]
So the newly generated libtorchbind_test.so isn't yet available when running the test. I'll check that out.

**Comment by jeffdaily** [comment]
I think you need to add cmake arg `INSTALL_TEST=ON` to the `.jenkins/pytorch/build.sh` file.  It appears in the `.jenkins/caffe2/build.sh` file.  That var seems to control whether `torchbind_test` gets installed.

**Comment by t-vi** [comment]
Turns out the jenkins setup is too cleanly and removes the .so from build/lib, but it seems I can use it from the installation directory instead.

**Comment by t-vi** [comment]
So I think this is good now.

**Comment by t-vi** [comment]
It runs the TorchBind tests and seems all OK. @malfet what Do you think?
```
15:38:44 test_profiler_custom_op (jit.test_torchbind.TestTorchbind) ... ok
15:38:44 test_torchbind (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_class_attribute (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_deepcopy (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_getstate (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_instantiate_missing_class (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_lambda_method (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_no_init (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_optional_explicit_attr (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_pickle_serialization (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_python_deepcopy (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_return_instance (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_return_instance_from_method (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_return_tuple (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_save_load (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_take_as_arg (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_take_instance_as_method_arg (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_tracing (jit.test_torchbind.TestTorchbind) ... ok
15:38:45 test_torchbind_tracing_nested (jit.test_torchbind.TestTorchbind) ... ok
```

File: test/cpp/jit/CMakeLists.txt [hunk 1]
```diff
@@ -97,8 +97,6 @@ elseif(USE_ROCM)
     ${PYTORCH_HIP_HCC_LIBRARIES}
     ${TORCH_CUDA_LIBRARIES})
 
-  target_link_libraries(test_jit PRIVATE caffe2_gpu)
```
Comment (walterddr) [inline test/cpp/jit/CMakeLists.txt:100]: qq: why getting rid of this one? is this removal somehow related to enabling torchbind test?

Comment (t-vi) [reply test/cpp/jit/CMakeLists.txt:100]: It's not required for the torchbind test, but so it is the twin of the line we need to remove for the torchbind test - the reason being that caffe2_gpu does not exist anymore.

Comment (t-vi) [reply test/cpp/jit/CMakeLists.txt:100]: Thank you for looking at this!

**Review by walterddr — APPROVED** [review]
